bitvm/src/hash/sha256_test_utils.rs

```diff
@@ -0,0 +1,60 @@
+use crate::hash::sha256::reference_sha256;
+use serde::Deserialize;
+use sha2::{Digest, Sha256};
+use std::error::Error;
+use std::fs::File;
+use std::io::BufReader;
+
+#[derive(Deserialize, Debug)]
+pub struct Message {
+    pub data: String,
+    pub count: u64,
+    pub note: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TestVector {
+    pub message: Message,
+    #[serde(rename = "SHA-256")]
+    pub sha256: String,
+}
+
+/// Read test vectors from JSON file
+pub fn read_sha256_test_vectors() -> Result<Vec<TestVector>, Box<dyn Error>> {
+    let path = "src/hash/sha256_official_test_vectors.json";
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+    let (_ignored, vectors): (serde::de::IgnoredAny, Vec<TestVector>) =
+        serde_json::from_reader(reader)?;
+    Ok(vectors)
+}
+
+/// Prepare test vector for use in testing. If the input is longer than 512 bytes, it will be hashed twice.
+pub fn prepare_test_vector(data: &str, count: u64, expected_hex: &str) -> (String, String) {
+    let mut input = data.repeat(count as usize).as_bytes().to_vec();
+    let mut expected = hex::decode(expected_hex).unwrap();
+
+    // Some of the test vectors are longer than 512 bytes, so we need to hash them twice
+    let double_hashing_needed = input.len() > 512;
```

Comment:
There's an off-by-one error here. The implementation will panic if the input length is 512 bytes.